### PR TITLE
feat: creates task manager app

### PR DESCRIPTION
**Pull request by pedrohenrique-ql**
## Features

- Creates `Task` entity with uuid, title, description, date and priority
- Creates getters and setters for `Task` class
- Creates `TaskList` entity with linked hash map of tasks
- Find task by id in `TaskList`
- Update task by id in `TaskList`
- Update task priority by id in `TaskList`
- Delete task by id in `TaskList`
- List tasks sorted by priority and due date in `TaskList`

### CODE REVIEW

File: task-manager/src/test/java/TaskTest.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    @BeforeEach
+    public void setUp() {
+        task = new Task("Title", "Description", LocalDate.now(), TaskPriority.HIGH);
+    }
+
+    @Test
+    public void testTaskCreation() {
+        assertNotNull(task);
+
```
Comment (jaugustorufino) [inline task-manager/src/test/java/TaskTest.java]: Is this Not Null check really necessary? I can see that you are testing some things later that make sure the task is not null.

File: task-manager/src/main/java/org/taskmanager/TaskList.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        return this.tasks.get(id);
+    }
+
+    public Task updateTask(UUID id, Task updatedTask) {
+        Task taskToUpdate = this.findTask(id);
+
+        taskToUpdate.setTitle(updatedTask.getTitle());
+        taskToUpdate.setDescription(updatedTask.getDescription());
+        taskToUpdate.setDueDate(updatedTask.getDueDate());
+        taskToUpdate.setPriority(updatedTask.getPriority());
```
Comment (jaugustorufino) [inline task-manager/src/main/java/org/taskmanager/TaskList.java]: In this case, can we just define the main attributes of a task? The taskToUpdate will have a method responsible for this, keeping the clean code. What do you think?

Comment (pedrohenrique-ql) [reply task-manager/src/main/java/org/taskmanager/TaskList.java]: Yeah, it's a good idea!

File: task-manager/src/test/java/TaskTest.java (hunk 1 above)
Comment (pedrohenrique-ql) [reply task-manager/src/test/java/TaskTest.java]: Yes, I'll remove this unnecessary test.

**Review by jaugustorufino — APPROVED** [review]
Good job!

**Pull request merged**